src/python/zquantum/qcbm/optimization_test.py

```diff
@@ -26,16 +27,16 @@ def test_qcbm_set_initial_params(self):
         num_qubits = 4
         topology = "all"
         epsilon = 1e-6
-        initial_params = generate_random_initial_params(num_qubits, seed=RNDSEED)
+        n_layers = 1
+        initial_params = generate_random_initial_params(num_qubits, n_layers=n_layers, topology=topology, seed=RNDSEED)
```

Comment:
Not totally relevant to this PR but made me remind that the `generate_random_initial_params` has a couple of other arguments, namely `min_val` and `max_val`, whose default values are pretty much arbitrary (0 and 1). This doesn't make too much sense to me, also because parameters are often angles, so it would be better to set their initial range to [-pi,pi] or something similar. Or maybe just don't define default values? @simonwa7 any suggestions?

Reply:
That might be a good idea for the QCBM, what do you think @simonwa7 ?

Reply:
@martamau Agreed. I think the qcbm will probably see an overhaul with the coming introduction of the `Ansatz` interface, so it should be addressed then.